adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/adapter/DynamoFeedDBPublisher.java

```diff
@@ -0,0 +1,218 @@
+        }
+
+        return categoriesList;
+    }
+
+    private String entryToString(Entry entry) {
+        final StringWriter writer = new StringWriter();
+
+        try {
+            entry.writeTo(writer);
+        } catch (IOException ioe) {
+            LOG.error("Unable to write entry to string. Unable to persist entry. Reason: " + ioe.getMessage(), ioe);
+
+            throw new PublicationException(ioe.getMessage(), ioe);
+        }
+
+        return writer.toString();
+    }
+
+    @Override
+    @NotImplemented
+    public AdapterResponse<Entry> putEntry(PutEntryRequest putEntryRequest) {
+        throw new UnsupportedOperationException("Not supported.");
+    }
+
+    @Override
+    @NotImplemented
+    public AdapterResponse<EmptyBody> deleteEntry(DeleteEntryRequest deleteEntryRequest) {
+        throw new UnsupportedOperationException("Not supported.");
+    }
+
+    /**
+     * To get the entry from dynamodb based upon two params?
+     *
+     * @param entryId:  It is the marker id for entry for every events
+     * @param feedName: feed name is used to search the records in dynamodb
+     * @return : list of entry found if that exits in dynamodb with the entryId and feedName.
+     */
+    private List<String> getEntry(String entryId, final String feedName) {
+        List<String> persistedEntriesObject = new ArrayList<String>();
+        Table table = dynamoDB.getTable(DynamoDBConstant.ENTRIES);
+        Index index = table.getIndex(DynamoDBConstant.ENTRY_ID_FEED_INDEX);
+        QuerySpec spec = new QuerySpec()
+                .withKeyConditionExpression("entryId = :entryId and feed = :feed")
+                .withValueMap(new ValueMap()
+                        .withString(":entryId", entryId)
+                        .withString(":feed", feedName));
+        ItemCollection<QueryOutcome> persistedEntryItems = index.query(spec);
+        Iterator<Item> itemsIterator = persistedEntryItems.iterator();
+        while (itemsIterator.hasNext()) {
+            Item item = itemsIterator.next();
+            persistedEntriesObject.add(item.toJSONPretty());
+        }
+        return persistedEntriesObject;
+    }
+
+    private void incrementCounterForFeed(String feedName) {
+
+        if (!counterMap.containsKey(feedName)) {
+            synchronized (counterMap) {
+                if (!counterMap.containsKey(feedName)) {
+                    Counter counter = Metrics.newCounter(DynamoFeedDBPublisher
+                            .class, "entries-created-for-" + feedName);
+                    counterMap.put(feedName, counter);
+                }
+            }
+        }
+
+        counterMap.get(feedName).inc();
+    }
+
+    /**
+     * Sets the date in String format as we save the date in string in dynamodb.
+     *
+     * @param date Date Object to be formatted in string.
+     * @return
+     */
+    private String getDateFormatInString(Date date) {
+        Format formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
Date format can be added to the Constants file.

Reply:
yes ..done